third_party/rust/third_party.toml

```diff
@@ -201,14 +201,25 @@ if (target_cpu == "arm64" &&
 '''
 
 [dependencies.curve25519-dalek]
-version = "3.2.1"
-features = [ "alloc", "serde", "std", "u64_backend" ]
+version = "4.1.1"
+allow-first-party-usage = false
+features = [ "rand_core", "serde" ]
+# curve25519-dalek-derive is only used on x64
+# gnrt did not detect that due to a complex build.rs condition
+gn-variables-lib = '''
+if (target_cpu == "x64") {
+  deps += [
+    "//brave/third_party/rust/cpufeatures/v0_2:lib", 
+    "//brave/third_party/rust/curve25519_dalek_derive/v0_1:lib",
+  ]
+}
+'''
 
 [dependencies.cpufeatures]
 version = "0.2"
 allow-first-party-usage = false
 gn-variables-lib = '''
-if (is_linux && target_cpu == "arm64") {
+if (target_cpu == "arm64") {
```

Comment:
I think libc is a no-op for windows, but just for consistency with Cargo.toml this dep should be `&& (is_android || is_apple || is_linux)`